pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -1268,7 +1268,17 @@ class Timedelta(_Timedelta):
 
             kwargs = {key: _to_py_int_float(kwargs[key]) for key in kwargs}
 
-            nano = convert_to_timedelta64(kwargs.pop('nanoseconds', 0), 'ns')
+            # GH43764, making sure any nanoseconds contributions from any kwarg
+            # is taken into consideration
+            nano = convert_to_timedelta64(
```

Comment:
you are entirely duplicating L1283

Reply:
My modifications removed "microseconds" till "seconds" from `kwargs`. Any other potential `kwarg`, such as `year` etc, are handled the same as the previous way, namely via L1283. Do you have other suggestions?

Reply:
i actually don't have a problem with doing what you are doing but then you need to entirely remove L1282-1284 and handle all the kwargs

 (well you still need the try/except).

Reply:
this will also slow this function down a lot. rather than poping be explicit on the argument handling

Reply:
Could you check the last changes? I profiled instantiation time and it is taking about 50% the time as in master.
```
def test():
	pd.Timedelta(seconds=float(106751 * 24 * 3600), nanoseconds=1)
	pd.Timedelta(minutes=-7)
	pd.Timedelta(seconds=1234e-9)
	pd.Timedelta(seconds=1e-9, milliseconds=1e-5, microseconds=1e-1)
	pd.Timedelta(days=1, seconds=1e-9, milliseconds=1e-5, microseconds=1e-1)
```